classes/client.php

```diff
@@ -127,21 +127,27 @@ public static function get_records_without_subscription() {
      * Get client by domain.
      *
      * @param string $domain
+     * @param string $clientid Optional to restrict access to scorm if domain is used by more than one client.
      * @return client
+     * @throws \dml_exception
      */
-    public static function get_record_by_domain(string $domain) {
+    public static function get_record_by_domain(string $domain, string $clientid) {
         global $DB;
 
-        $sql = "SELECT client.*
-                  FROM {scormremote_clients} client
-                 WHERE EXISTS (
-                           SELECT 1
-                             FROM {scormremote_client_domains} dom
-                            WHERE client.id = dom.clientid
-                              AND dom.domain = :domain
-                       )";
+        $clientidclause = ($clientid) ? "AND client.id = :clientid" : "";
 
-        $record = $DB->get_record_sql($sql, ['domain' => $domain]);
+        $sql = "SELECT client.*
+                FROM   mdl_scormremote_clients AS client
```

Comment:
can you please format this according to the moodle sql code standards, more or less right align all the bits in CAPS

The subselect can be indented a bit less to make it more readable